### ISSUE TITLE
Minio demote of one or more pool are not compatible with Veeam or any software with Immutability of objects.

### ISSUE THREAD

**Issue by tdsn0007** [minio/minio#17154]
## NOTE
If this case is urgent, please subscribe to [Subnet](https://min.io/pricing) so that our 24/7 support team may help you faster.

Minio cluster using Immutability of objects and data hosted on pool who we wants to decomiss are not possible. Because metadata of immutability used by veeam are not updated on nodes started to decomiss.

## Expected Behavior

Updates metada inside nodes in process to decomiss.

## Current Behavior

Metadata are blocked on decomissing nodes and veeam show error because can´t update using PutOblectLock retention.

## Possible Solution

Maybe permit updates metadata on nodes who start to decossioning

## Steps to Reproduce (for bugs)

1. Create an backup using veeam and objectlock with immutability on one cluster with one or more nodes.
2.Create another pool
3.Migrate all data of old pool using decomissing of pool who has veeam data.
4.If the process are longer than days (big backups always) we have error on veeam when they start to update metadata of ObjectLock

## Context

Using veeam and remove old pools, very real situation. 

## Regression

No
## Your Environment

* Version used (`minio --version`): Latest
* Server setup and configuration: 2 nodes pool + 4 new nodes pool
* Operating System and version (`uname -a`): Red Hat 9.1

**Comment by harshavardhana** [comment]
@tdsn0007 this is correct at the moment its not allowed, I am working on a fix to allow Veeam like tools to update metadata alone.

**Comment by harshavardhana** [comment]
> 1. 3.Migrate all data of old pool using decomissing of pool who has veeam data.
>    4.If the process are longer than days (big backups always) we have error on veeam when they start to update metadata of ObjectLock

Can you share the error? @tdsn0007

**Comment by tdsn0007** [comment]
> > 1. 3.Migrate all data of old pool using decomissing of pool who has veeam data.
> >    4.If the process are longer than days (big backups always) we have error on veeam when they start to update metadata of ObjectLock
> 
> Can you share the error? @tdsn0007

Unfortunally not now. Because when we see this problem, we started to migrate all data (months and months...).

**Comment by harshavardhana** [comment]
> > > 1. 3.Migrate all data of old pool using decomissing of pool who has veeam data.
> > >    4.If the process are longer than days (big backups always) we have error on veeam when they start to update metadata of ObjectLock
> > 
> > 
> > Can you share the error? @tdsn0007
> 
> Unfortunally not now. Because when we see this problem, we started to migrate all data (months and months...).

Well I tried to reproduce I can see that I can set retention on objects that are part of the pool that is being decommissioned.

Please share the error if you have, I am closing this issue as working properly as per my testing.

**Comment by tdsn0007** [comment]
> > > > 1. 3.Migrate all data of old pool using decomissing of pool who has veeam data.
> > > >    4.If the process are longer than days (big backups always) we have error on veeam when they start to update metadata of ObjectLock
> > > 
> > > 
> > > Can you share the error? @tdsn0007
> > 
> > 
> > Unfortunally not now. Because when we see this problem, we started to migrate all data (months and months...).
> 
> Well I tried to reproduce I can see that I can set retention on objects that are part of the pool that is being decommissioned.
> 
> Please share the error if you have, I am closing this issue as working properly as per my testing.

Hi, Veeam can change retention, but are hard codded for 10 days + retention. If you change, only 10 days after the change they will change the metadata.

**Comment by harshavardhana** [comment]
> > > > > 1. 3.Migrate all data of old pool using decomissing of pool who has veeam data.
> > > > >    4.If the process are longer than days (big backups always) we have error on veeam when they start to update metadata of ObjectLock
> > > > 
> > > > 
> > > > Can you share the error? @tdsn0007
> > > 
> > > 
> > > Unfortunally not now. Because when we see this problem, we started to migrate all data (months and months...).
> > 
> > 
> > Well I tried to reproduce I can see that I can set retention on objects that are part of the pool that is being decommissioned.
> > Please share the error if you have, I am closing this issue as working properly as per my testing.
> 
> Hi, Veeam can change retention, but are hard codded for 10 days + retention. If you change, only 10 days after the change they will change the metadata.

I tested that and it works fine. Please share the actual error that you see. Until then I consider this issue to be resolved for now.
